### PR TITLE
fix(datepicker): fixed timezone offset bug

### PR DESCRIPTION
**Pull request by mario-subo**
offsets the timezone from the date to fix the bug where the day would be set wrongly around midnight

### CODE REVIEW

**Review by yousifalraheem — APPROVED** [review]

File: lib/src/Datepicker/Datepicker.tsx [hunk 1]
```diff
@@ -57,7 +57,12 @@ export const Datepicker: React.FunctionComponent<DatepickerProps> = ({
 
     const getStringFromDate = React.useCallback(
         (d: Date, monthPicker: boolean): string => {
-            return isValidDate(d) ? d?.toISOString()?.substr(0, monthPicker ? 7 : 10) || "" : "";
+            if (isValidDate(d)) {
+                const tzoffset: number = new Date().getTimezoneOffset() * 60000;
```
Comment (omarVodiaK) [inline lib/src/Datepicker/Datepicker.tsx:61]: are you converting to a different timezone here ?

File: lib/src/Datepicker/Datepicker.tsx [hunk 2]
```diff
@@ -107,7 +103,8 @@ export const Datepicker: React.FunctionComponent<DatepickerProps> = ({
 
     const handleChangeCustomDay = (e: React.ChangeEvent<HTMLInputElement>): void => {
         if (!monthPicker) {
-            setCustomDay(Number(e.target.value));
+            const v: number = e.target?.value && !Number.isNaN(e.target?.value) ? Number(e.target.value) : null;
```
Comment (omarVodiaK) [inline lib/src/Datepicker/Datepicker.tsx:106]: i think you should check if e.target.value is a number before using it inside Number.isNan(e.traget.value)
```suggestion
            const v: number = e.target?.value && !Number.isNaN(Number(e.target.value)) ? Number(e.target.value) : null;
```

File: lib/src/Datepicker/Datepicker.tsx (hunk 1 above)
Comment (mario-subo) [reply lib/src/Datepicker/Datepicker.tsx:61]: No I'm adjusting the date in a way which removes any timezone offset

File: lib/src/Datepicker/Datepicker.tsx (hunk 2 above)
Comment (mario-subo) [reply lib/src/Datepicker/Datepicker.tsx:106]: good catch, thanks!! I'll update it

File: lib/src/Datepicker/Datepicker.tsx [hunk 3]
```diff
@@ -69,26 +74,18 @@ export const Datepicker: React.FunctionComponent<DatepickerProps> = ({
...
+            if (!max || (max && d <= max)) {
+                return true;
             } else {
-                reject();
+                return false;
             }
-        });
+        } else {
+            return false;
+        }
```
Comment (omarVodiaK) [inline lib/src/Datepicker/Datepicker.tsx:88]:
```javascript
if (!min && !max) {
  return true;
} else if (min && d >= min) {
   if (!max || (max && d <= max)) {
       return true;
   }
}
return false;
```

File: lib/src/Datepicker/Datepicker.tsx (hunk 1 above)
Comment (omarVodiaK) [reply lib/src/Datepicker/Datepicker.tsx:61]: so when a user selects a date would there be any information on what timezone that date has ?

**Review by omarVodiaK — APPROVED** [review]

File: lib/src/Datepicker/Datepicker.tsx (hunk 1 above)
Comment (mario-subo) [reply lib/src/Datepicker/Datepicker.tsx:61]: If the user passes a date with timezone info in it, the timezone info will be removed and only the date will be respected.

Comment (omarVodiaK) [reply lib/src/Datepicker/Datepicker.tsx:61]: i think it's a bit dangerous to do that, especially if you are adding data in a timezone and the data is saved in a server that have a different timezone

Comment (mario-subo) [reply lib/src/Datepicker/Datepicker.tsx]: Okay I actually found a better way of achieving this so I updated this part. No timezone modification needed.

**Pull request merged**